### ISSUE TITLE
Do Geometry objects support Z and M?

### ISSUE THREAD

**Issue by cmoravec** [Esri/runtime-questions#48]
As a followup will the geometry engine support calculations with Z and M?

**Comment by shellygill** [comment]
Can you tell us more about what calculations you want? Geometries in Quartz support Z and M values at Beta 1, and we will also add functions in a following beta so you can match geometry z/m awareness with geodatabase table awareness when editing, but beyond that we'd be interested to hear from users what their priorities are, and more about the scenarios where they'd be used.

**Comment by madmapper** [comment]
Not attending the ESRI US this year, but I can chime in with one of our use-cases that require Z values.  We currently support field observation using an app that does position offset based on measurement values entered by the user.  (For instance using reticle binoculars a user might enter a bearing and reticle count from the horizon)  

We currently use the system (iOS) location service to get elevation and a field in the feature table to store the value to use in the offset calculations since the iOS SDK does not offer a Z value property in the AGSLocation object (If Z were supported in the offline geodatabase we could store this information spatially).  But at this point the offset is really only being done in X/Y space.

**Comment by shellygill** [comment]
Thanks @madmapper - it sounds like your request is mainly for the Location object to have a Z value, and there's no specific z-value calculations that you're asking for here. I see you previously asked about this in issue #5 . Ideally the location API would include height information if the underlying platform location service passes on height information, for example represented as a Point geometry with a Z value - I will see if I can find out any more information or get confirmation of that.

**Comment by cmoravec** [comment]
Ideally I would like to have fully support for linear referencing operations (measure to coordinate, coordinate to measure, etc.).  However, my basic need is to ensure that if the geometry does have M values that any geometry operations that should obey M values, do.  For example, if I have a polyline with M values and I clip that polyline using a polygon, the newly created endpoints of the polyline should have calculated measures based on the spatial location of the cut.  This appears to work already as when I tested a clip operation in the geometry engine, I got measures back at the new endpoints of the polyline, and they seemed correct.

As for Z values, we do live in a 3D world ;) So it would be nice if calculations took into account Z values correctly, for example distance calculations between two points that contain Z values should be done in 3D.  Basically these should be honored just as the measures are, if the Z value can be used in a calculation then it should (or at least I should have the option as a developer to get that type of calculation).
